src/type.js

```diff
@@ -12,10 +13,30 @@ import {getField} from './field';
  * @return {Object} types
  */
 function getTypes (models) {
-  var types = reduce(models, (types, model) => {
+
+  var types = {};
+  var typeNames = Object.keys(models);
+  var {nodeInterface, nodeField} = nodeDefinitions(
+    (globalId) => {
+      var {type, id} = fromGlobalId(globalId);
+      if (typeNames.indexOf(type) > -1) {
+        return models[type].model.findById(id).then((mongooseObj) => {
+          var obj = mongooseObj.toObject();
+          obj._graphqlType = type;
+          return obj;
+        });
+      }
+      return null;
+    }, (obj) => {
+      return types[obj._graphqlType];
```

Comment:
I'm guessing the thinking is that the type resolver function closes over `var types` so it's available (and filled) when it's invoked later. `nodeField` is not accessed until it's returned from this function, after `types` is populated.